### PR TITLE
Update to use group id in artifact name when using maven 3 or higher

### PR DESCRIPTION
**Pull request by pnickoll**
When using Maven 3 or higher the output filename for WARS includes the group id. Updated m2e to reflect this change in Maven.

### CODE REVIEW

**Comment by mickaelistria** [comment]
@fbricon can you please review?

**Comment by fbricon** [comment]
I skimmed at the code, that looks fine, but I noticed the build is broken. Haven't had time to fix it yet. If you want to look at updating the target platform in the process, feel free to do so.

**Comment by fbricon** [comment]
@pnickoll does the build pass locally for you? Jenkins no likey

> [ERROR] Plugin org.codehaus.mojo:build-helper-maven-plugin:1.3 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.codehaus.mojo:build-helper-maven-plugin:jar:1.3: 1 problem was encountered while building the effective model
[ERROR] [FATAL] Non-parseable POM /home/jenkins/.m2/repository/org/codehaus/mojo/build-helper-maven-plugin/1.3/build-helper-maven-plugin-1.3.pom: UTF-8 BOM plus xml decl of ISO-8859-1 is incompatible (position: START_DOCUMENT seen <?xml version="1.0" encoding="ISO-8859-1"... @1:42)  @ line 1, column 42
[ERROR] -> [Help 1]

Can you try updating the build-helper-maven-plugin version?

**Comment by fbricon** [comment]
For some reason m2e fails to initialize properly during CI tests https://ci.eclipse.org/m2e/job/m2e-wtp/job/PR-28/11/artifact/org.eclipse.m2e.wtp.tests/target/work/data/.metadata/.log

Tests run fine on my Mac and Linux box.

🙄

**Comment by pnickoll** [comment]
The test ran fine for me on Windows too. FYI: I have a debug commit in there - I will remove it before merging

File: org.eclipse.m2e.wtp.jaxrs/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %Bundle-Name
 Bundle-SymbolicName: org.eclipse.m2e.wtp.jaxrs;singleton:=true
-Bundle-Version: 1.6.0.qualifier
```
Comment (mickaelistria) [inline org.eclipse.m2e.wtp.jaxrs/META-INF/MANIFEST.MF:5]: Why this change here? Only modifies bundles should have their version bumped (if not already done since last release).

**Comment by fbricon** [comment]
> getExecutionEnvironment

That might be possible but I haven't seen that exception in the latest failures. Where did you see it?

File: org.eclipse.m2e.wtp.jaxrs/META-INF/MANIFEST.MF (hunk 1 above)
Comment (pnickoll) [reply org.eclipse.m2e.wtp.jaxrs/META-INF/MANIFEST.MF:5]: @mickaelistria I (incorrectly by the sounds of things) assumed you'd want to keep the versions of all the bundles in step. I can change just the bundle I am modifying.

**Comment by pnickoll** [comment]
@fbricon I saw the getExecutionEnvironment in https://ci.eclipse.org/m2e/job/m2e-wtp/job/PR-28/11/console however it seems your update to pull in m2e 2.3 fixed the issue! Thank you!

File: org.eclipse.m2e.wtp.jaxrs/META-INF/MANIFEST.MF (hunk 1 above)
Comment (pnickoll) [reply org.eclipse.m2e.wtp.jaxrs/META-INF/MANIFEST.MF:5]: @mickaelistria Understood, I'll update that.

**Comment by pnickoll** [comment]
@mickaelistria With respect to increasing versions - this build (https://ci.eclipse.org/m2e/job/m2e-wtp/job/PR-28/15/console) is implying I need to increase the version - although I've not modified that bundle?

**Comment by mickaelistria** [comment]
I see that the build from master is very broken right now; that's most likely the reason why you deal with so much issues with versions. They're not related to your patch.
I will try to make the build work better, and that will probably involve fixing some versions, then we will be able to rebase your patch so we can focus on the actual payload change without being too bothered by versions.

**Comment by mickaelistria** [comment]
So I actually merged it by mistake while trying to make master build works. So while it's not a great move per se, at least, we can consider this PR closed ( https://github.com/eclipse-m2e/m2e-wtp/commit/994696de6523dac55bc5b58a1c4e088da0bbe83d ) ;)

**Comment by pnickoll** [comment]
@mickaelistria Thanks for merging (even if in error :) ), did you manage to get the builds working again?

**Comment by mickaelistria** [comment]
https://ci.eclipse.org/m2e/job/m2e-wtp/job/master/8/ should soon complete as yellow/unstable (for an infra reason) and the content will then be available in the snapshots p2 repository.

**Comment by pnickoll** [comment]
@mickaelistria Do you know when this will be pushed to a released version and when we can expect it in an Eclipse release?

**Comment by fbricon** [comment]
Later today if everything goes well.

**Comment by fbricon** [comment]
m2e-wtp 1.6.1 is available from: 
- https://download.eclipse.org/technology/m2e/m2e-wtp/releases/1.6.1/
- https://download.eclipse.org/technology/m2e/m2e-wtp/releases/latest/

Simrel PR: https://github.com/eclipse-simrel/simrel.build/pull/44
